lib/stencil-bundle.js

```diff
@@ -151,51 +152,98 @@ class Bundle {
         };
     }
 
-    assembleTemplatesTask(callback) {
+    async assembleTemplatesTask(callback) {
         console.log('Template Parsing Started...');
+        const internalTemplatesList = await recursiveReadDir(this.templatesPath, ['!*.html']);
+        const searchExternalLib = (content) => {
+            const externalPathRegex = /{{2}>\s*(external)[^{]*?}{2}/g;
+            const externalTemplatesImports = content.match(externalPathRegex);
+            if (!externalTemplatesImports) return [];
 
-        recursiveReadDir(this.templatesPath, ['!*.html'], (readdirError, files) => {
-            if (readdirError) {
-                return callback(readdirError);
-            }
-
-            const partials = files.map((file) => {
-                return upath.toUnix(
-                    file.replace(this.templatesPath + path.sep, '').replace(/\.html$/, ''),
-                );
-            });
+            return externalTemplatesImports.map((_import) => _import.split('/')[1]);
+        };
+        const checkTemplate = async (templatePath) => {
+            const content = await fs.promises.readFile(templatePath, { encoding: 'utf-8' });
+            const res = searchExternalLib(content);
 
-            return async.map(
-                partials,
-                templateAssembler.assembleAndBundle.bind(null, this.templatesPath),
-                (err, results) => {
-                    const ret = {};
+            return res;
+        };
 
-                    if (err) {
-                        return callback(err);
-                    }
+        let externalLibs;
+        try {
+            externalLibs = (await async.map(internalTemplatesList, checkTemplate)).flat();
```

Comment:
I would recommend not to use `async` library, where you can implement it in vanilla js

Reply:
done